static-analysis/autodispose-lint/src/test/kotlin/com/uber/autodispose/lint/AutoDisposeDetectorTest.kt

```diff
@@ -1052,4 +1080,103 @@ class AutoDisposeDetectorTest {
         .run()
         .expectClean()
   }
+
+  @Test fun withScope_withScopeProvider_missingAutoDispose_shouldError() {
+    lint()
+        .files(rxJava2(),
+            SCOPE_PROVIDER,
+            AUTODISPOSE_CONTEXT,
+            WITH_SCOPE_PROVIDER,
+            kotlin("""
+          package foo
+          import io.reactivex.Observable
+          import com.uber.autodispose.ScopeProvider
+          import com.uber.autodispose.withScope
+          class ExampleClass {
+            lateinit var scopeProvider: ScopeProvider
+            fun names() {
+              val observable = Observable.just(1)
+              withScope(scopeProvider) {
+                observable.subscribe()
+              }
+            }
+          }
+        """).indented())
+        .issues(AutoDisposeDetector.ISSUE)
+        .run()
+        .expectErrorCount(1)
+  }
+
+  @Test fun withScope_withCompletable_missingAutoDispose_shouldError() {
+    lint()
+        .files(rxJava2(),
+            AUTODISPOSE_CONTEXT,
+            WITH_SCOPE_COMPLETABLE,
+            kotlin("""
+          package foo
+          import io.reactivex.Completable
+          import io.reactivex.Observable
+          import com.uber.autodispose.withScope
+          class ExampleClass {
+            fun names() {
+              val observable = Observable.just(1)
+              withScope(Completable.complete()) {
+                observable.subscribe()
```

Comment:
What happens if you have another line before this `subscribe()`? 

Let's say
```kotlin
val a = 1
observable.subscribe()
```

Just wondering if the `subscribe` has to be the only code in the LambdaExpression for lint to detect correctly.

Reply:
Still works! The visitor will visit it like any other `UBlockExpression`. This is similar to if, say, lint runs on the contents of a `run { }` block or function body.

Reply:
Cool, sounds good! Let's merge!